receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go

```diff
@@ -90,7 +90,23 @@ func (s *scraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
 	}
 
 	usages := make([]*deviceUsage, 0, len(partitions))
+
+	type mountKey struct {
+		mountpoint string
+		device     string
+	}
+	seen := map[mountKey]struct{}{}
+
 	for _, partition := range partitions {
+		key := mountKey{
+			mountpoint: partition.Mountpoint,
+			device:     partition.Device,
+		}
+		if _, ok := seen[key]; partition.Mountpoint != "" && ok {
```

Comment:
Can you clarify to me why the check for Mountpoint being an empty string is here?

Reply:
The "Virtual partitions" test fails without that as virtual partitions don't have mount points. See test `Include device filtering that includes virtual partitions`  (I'm not really convinced about the validity of that particular feature/test, you get mount entries without devices, where only a filesystem type like procfs is used, but they do get mounted somewhere)